### DISCUSSION TITLE
Should "go to previous buffer" / "go to next buffer" behaviour be changed?

### DISCUSSION THREAD

**Discussion by dariooddenino** [helix-editor/helix#4502] [Ideas] [answered]
The current implementation of `gp`/`gn` uses the buffer id, and so the order on which the buffers were opened.

This imho becomes quickly unusable, especially when working with split windows and multiple buffers.

I think it would be better if the commands worked on an ordering based on "last visualized" buffer.

**Comment by helmesjo** [comment]
Yes. The way it is now is very confusing and usually I end up just spinning through buffers until I reach the one I want (without knowing when to expect it).

**Comment by getreu** [comment]
> The way it is now is very confusing and usually I end up just spinning through buffers until I reach the one I want

Have you tried `[space] b`?

**Reply by helmesjo** [reply]
Yes, absolutely (but it's similar in that the order is not very intuitive so it has the same issue in my opinion). `ga` is my go-to, and if cycling the buffers was (optionally) sorted in last-viewed order that would be what my brain "expects" (order opened is nothing that one keeps track of).

**Comment by archseer** [comment, marked as answer]
> "last visualized" buffer.

You can already use the jumplist, `ga` or `gm` for this. `gn`/`gp` is deliberately meant to have a stable order similar to `:bn`/`:bp` in Vim and follows the order of the buffer list.
